src/query/expression/src/evaluator.rs

```diff
@@ -951,13 +951,96 @@ impl<'a> Evaluator<'a> {
         unreachable!("expr is not a set returning function: {expr}")
     }
 
+    fn run_array_fold(&self, column: &Column, expr: &Expr) -> Result<Value<AnyType>> {
+        if column.data_type().is_null() || column.len() < 1 {
+            return Ok(Value::Scalar(Scalar::Null));
+        }
+
+        let first_element = column.index(0).unwrap();
+        let cast_value = self.run_cast(
```

Comment:
We can add a `ScalarExpr::CastExpr` for the argument in the `resolve_lambda_function` to convert the original column data to the target type without having to explicitly call the `run_cast` here